client/src/cordova/apple/OutlineAppleLib/Sources/OutlineError/OutlineError.swift

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Tun2socks
+
+/// Lists all errors that might be thrown in Swift code.
+public enum OutlineError: Error {
+  case internalError(message: String)
+  case vpnPermissionNotGranted(cause: Error)
+  case setupSystemVPNFailed(cause: Error)
+}
+
+/// Helper init to convert from an OutlineError to DetailedJsonError.
+public extension DetailedJsonError {
```

Comment:
Why not a conversion function? Extensions are obscure

Reply:
Converted to `asDetailedJsonError` function.